### ISSUE TITLE
How can enum descriptions be supported?

### ISSUE THREAD

**Issue by Janpot** [StefanTerdell/zod-to-json-schema#63]
```tsx
import * as z from "zod";
import { zodToJsonSchema } from "zod-to-json-schema";

const mySchema = z.union([
  z.literal("foo").describe("I'm a foo"),
  z.literal("bar").describe("I'm a bar"),
]);

const jsonSchema = zodToJsonSchema(mySchema);

console.log(JSON.stringify(jsonSchema, null, 2));
```

results in

```json
{
  "type": "string",
  "enum": [
    "foo",
    "bar"
  ],
  "$schema": "http://json-schema.org/draft-07/schema#"
}
```

The descriptions are lost.

Would it make sense to instead implement this as `anyOf`?

```json
{
  "type": "string",
  "anyOf": [
    { "const": "foo", "description": "I'm a foo" },
    { "const": "bar", "description": "I'm a bar" }
  ],
  "$schema": "http://json-schema.org/draft-07/schema#"
}
```

**Comment by StefanTerdell** [comment]
We could check for meta in the union parser and avoid collapsing if its present.

**Comment by StefanTerdell** [comment]
Released in 3.21.1
